plugins/system/cookiemanager/cookiemanager.php

```diff
@@ -0,0 +1,342 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  System.cookiemanager
+ *
+ * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Uri\Uri;
+use Joomla\CMS\WebAsset\WebAssetManager;
+use Joomla\CMS\Component\ComponentHelper;
+
+/**
+ * System plugin to manage cookies.
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class PlgSystemCookiemanager extends CMSPlugin
+{
+	/**
+	 * Application object.
+	 *
+	 * @var    JApplicationCms
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $app;
+
+	/**
+	 * For cookie banner
+	 *
+	 * @var    CookieBanner
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $cookieBanner;
+
+	/**
+	 * For preferences banner
+	 *
+	 * @var    Preferences
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $preferences;
+
+	/**
+	 * Database object
+	 *
+	 * @var    DatabaseDriver
+	 * @since  __DEPLOY_VERSION__
+	 */
+	 protected $db;
+
+	 /**
+	  * For script in DOM
+	  *
+	  * @var    Script
+	  * @since  __DEPLOY_VERSION__
+	  */
+	 protected $script = [];
+
+	 /**
+	  * For scripts in DOM
+	  *
+	  * @var    Scripts
+	  * @since  __DEPLOY_VERSION__
+	  */
+	 protected $scripts;
+
+	 /**
+	  * For cookie category
+	  *
+	  * @var    Category
+	  * @since  __DEPLOY_VERSION__
+	  */
+	 protected $category;
+	/**
+	 * Add assets for the modal.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+
+	 */
+	public function onBeforeCompileHead()
+	{
+		ob_start();
+		ob_implicit_flush(false);

```

Comment:
Why is this needed?

Reply:
because we need to render cookie setting scripts 
So first capture the output buffer, concatenate with scripts and then render it (get idea from debug plugin)